gitdiff/patch_header.go

```diff
@@ -369,3 +373,61 @@ func parseHeaderMail(mailLine string, r io.Reader) (*PatchHeader, error) {
 
 	return h, nil
 }
+
+func cleanupSubject(s string) string {
+	r := []rune(s)
+
+	// This is meant to be compatible with
+	// https://github.com/git/git/blob/master/mailinfo.c:cleanup_subject().
+	// If compatibility with `git am` drifts, go there to see if there
+	// are any updates.
+
+	var at int
+	for at = 0; at < len(r); {
+		switch r[at] {
+		case 'r', 'R':
+			// Detect re:, Re:, rE: and RE:
+			if at+3 > len(r) {
+				break
+			}
+			if (r[at+1] == 'e' || r[at+1] == 'E') &&
+				r[at+2] == ':' {
```

Comment:
style: I think I'd combine these into a single condition:

```suggestion
			if at+2 < len(r) && (r[at+1] == 'e' || r[at+1] == 'E') && r[at+2] == ':' {
```

Reply:
Ack